department-of-veterans-affairs/vets-website
# Multiple page dependent children

## Summary
Implement the Multiple Page Response pattern on the Pension Benefits form (21P-527EZ) for dependent children

### Description
Replace the dependent children pages with the [Array Builder Pattern (Multiple responses list & loop)](https://design.va.gov/patterns/ask-users-for/multiple-responses#how-to-design-and-build---multi-page) pattern and [Design](https://www.figma.com/design/9JKK5Eo43uJWEr66JPiebc/Pension-Claim-Form-21P-527EZ?node-id=5132-46077&t=ZHdkWtU3FvopTYjA-0) reference

## Issue
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/82850

## How to run in local environment
1. Check out this branch locally
2. In your local environment, set the `pension_multiple_page_response` flipper to 'enabled' at http://localhost:3000/flipper/features/pension_multiple_page_response
3. Run `vets-website` and `vets-api`
4. Go to `http://localhost:3001/pension/apply-for-veteran-pension-form-21p-527ez/` in your browser

## How to verify
1. Continue to the dependents pages in the 'Household information' chapter (/household/dependents/summary)
2. Add, edit and delete a dependent via the multiple page list loop pattern
3. Verify that the UI matches the Design
4. Verify the layout on desktop and mobile viewports
5. Verify that the form data is saved to the api and redux store

## How to test
### Dependent Children Pages
- [ ] Add a dependent between 18-23 years and verify that the attending college page displays
- [ ] Add a dependent that does not live with you and verify that the address page displays
- [ ] Verify that the 'Is your child in school?' alert dynamically displays
- [ ] Verify that the 'Is your child seriously disabled?' alert dynamically displays
- [ ] Verify that the 'Are they currently married?' question dynamically displays when your child has been married
- [ ] Verify that field types are validated
- [ ] Verify that required fields are required

### Dependent Children Summary Page
- [ ] Do not add a dependent from the '/pensions/dependents/summary' page
- [ ] Add a dependent from the '/pensions/dependents/summary' page
- [ ] Cancel adding a dependent from any subsequent page after the '/pensions/dependents/summary' page and verify that the dependent is not added
- [ ] Add another(multiple) dependent from the '/pensions/dependents/summary' page
- [ ] Edit a dependent from the '/pensions/dependents/summary' page and verify that the changes are saved and an alert is displayed after the dependent is edited
- [ ] Cancel editing a dependent from any subsequent page after the '/pensions/dependents/summary' page and verify that changes are not saved
- [ ] Delete a dependent and verify that that an alert is displayed after the dependent is deleted

### Review and Submit Page
- [ ] On the '/review and submit' page, expand the 'Household information' chapter accordion and add a dependent
- [ ] Cancel adding a dependent from any subsequent page from the '/review-and-submit' page and verify that the dependent is not added
- [ ] Add another(multiple) dependent from the '/review and submit' page
- [ ] Edit a dependent from the 'review and submit' page and verify that the changes are saved and an alert is displayed after the dependent is edited
- [ ] Cancel editing a dependent from any subsequent page after the 'review and submit' page and verify that changes are not saved
- [ ] Delete a dependent and verify that that an alert is displayed after the dependent is deleted
- [ ] Verify that the form cannot be submitted when a dependent is incomplete

## Screenshots
| Summary | Add | Added |
| ------ | ------ | ------ |
|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_form-saved (11)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/fef76791-b435-4cf3-86d9-b223a62a931e)|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_form-saved (12)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/c62d756c-3889-4d1d-8363-ab4cc6a8c495)|![localhost_3001_pension_apply-for-veteran-pension-form-21p-527ez_form-saved (10)](https://github.com/department-of-veterans-affairs/vets-website/assets/7394764/99aabb84-dce0-4308-87b0-70f100a52b87)|

## What areas of the site does it impact?
Pension Benefits Application

### Quality Assurance & Testing

### Error Handling

### Authentication
